### PR TITLE
Switch from `poetry` to `uv`

### PR DESCRIPTION
**Pull request by knutwannheden**

### CODE REVIEW

File: rewrite/pyproject.toml [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+    "more-itertools>=10.4.0",
+    "openrewrite-remote",
+    "pip>=24.3.1",
+    "pypi-simple>=1.6.1",
 ]
+requires-python = ">=3.12, <4"
 
-[tool.poetry.dependencies]
-python = ">=3.9"
-pytest = "^8.3.2"
```
Comment (zieka) [inline rewrite/pyproject.toml:13]: I thought we were standardizing around 3.9 for some reason that now escapes me

File: rewrite/pyproject.toml [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
-    { include = "rewrite" }
+version = "0.0.0" # Updated dynamically during release
+authors = [{ name = "Moderne Inc.", email = "support@moderne.io" }]
+license = { text = "Moderne, Inc. Commercial License" }
+dependencies = [
+    "pytest>=8.3.4",
+    "more-itertools>=10.4.0",
+    "openrewrite-remote",
+    "pip>=24.3.1",
+    "pypi-simple>=1.6.1",
```
Comment (zieka) [inline rewrite/pyproject.toml:16]: probably dont need these; I believe their use is specific to remoting project

**Comment by zieka** [comment]
I am running into name space collisions when running test because rewrite module is overloaded:
```
    from rewrite import InMemoryExecutionContext, ParserInput, ParserBuilder, random_id, ParseError, ParseExceptionResult, \
E   ImportError: cannot import name 'InMemoryExecutionContext' from 'rewrite' (unknown location)
```

If I change the imports to rewrite_remote and link to my local project then tests go form 60 errors (imports) to 1 error (unrelated to imports)

**Comment by timtebeek** [comment]
Would this workflow need to be changed as well? 
https://github.com/openrewrite/rewrite-python/blob/8ec4473a32cd093228ad01c1039c2afa75d983df/.github/workflows/ci.yml#L45

File: rewrite/pyproject.toml (hunk 1 above)
Comment (knutwannheden) [reply rewrite/pyproject.toml:13]: Yes. I accidentally used 3.12 API to parse f-strings. As soon as we have a customer requiring an older Python interpreter, I will try to fix that.

File: rewrite/pyproject.toml (hunk 2 above)
Comment (knutwannheden) [reply rewrite/pyproject.toml:16]: Sounds good. I indeed just copied this.

**Pull request merged**